crates/iceberg/src/spec/table_metadata.rs

```diff
@@ -654,6 +670,33 @@ impl TableMetadata {
     }
 }
 
+/// Unnormalized table metadata, used as an intermediate type
+/// to build table metadata in a declarative way.
+#[derive(Debug, PartialEq, Deserialize, Eq, Clone)]
+pub struct UnnormalizedTableMetadata(TableMetadata);
+
+impl UnnormalizedTableMetadata {
+    /// Try to normalize the table metadata.
+    pub fn try_normalize(self) -> Result<TableMetadata> {
+        let mut metadata = self.0;
+        metadata.try_normalize()?;
+        Ok(metadata)
+    }
+}
+
+impl UnnormalizedTableMetadata {
+    /// Build a new [`UnnormalizedTableMetadata`] using the [`TableMetadataDeclarativeBuilder`].
+    pub fn builder() -> TableMetadataDeclarativeBuilder {
```

Comment:
unused? remove?

Reply:
This is a library and the constructor can be used outside of the crate. I like to attach some constructor to a type if available.
As it's an auxiliary type, I am OK with removing it as well.